### PR TITLE
Added Support for 1.20 - 1.20.2

### PR DESCRIPTION
**Pull request by LedariaTim1**

### CODE REVIEW

File: src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java [hunk 1]
```diff
@@ -85,7 +85,13 @@ public void onPacketSending(PacketEvent e) {
 			} else if (type == PacketType.Play.Server.RESPAWN) {
 				set.clear();
 			} else {
-				ChunkCoord coord = new ChunkCoord(packet.getIntegers().read(0), packet.getIntegers().read(1));
+                ChunkCoord coord;
+                if (NMSUtil.getMinorVersion() >= 20 && NMSUtil.getReleaseVersion() > 1) {
+                    coord = new ChunkCoord(packet.getStructures().read(0).getIntegers().read(0), packet.getStructures().read(0).getIntegers().read(1));
+
```
Comment (saharNooby) [inline src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java]:
```suggestion
```

File: src/main/java/me/saharnooby/plugins/leadwires/tracker/ProtocolUtil.java [hunk 2]
```diff
@@ -117,6 +116,9 @@ private static int getSilverfishId() {
 		switch (minorVersion) {
 			case latestVersion:
 				return latestId;
+
```
Comment (saharNooby) [inline src/main/java/me/saharnooby/plugins/leadwires/tracker/ProtocolUtil.java]:
```suggestion
```

File: src/main/java/me/saharnooby/plugins/leadwires/tracker/ProtocolUtil.java [hunk 3]
```diff
@@ -104,11 +104,10 @@ private static WrappedDataWatcher.WrappedDataWatcherObject getByteObject() {
 	private static int getSilverfishId() {
 		// https://wiki.vg/Entities
 		// https://wiki.vg/Pre-release_protocol#Entity_Metadata
-		final int latestId = 80;
-		final int latestVersion = 19;
+		final int latestId = 85;
+		final int latestVersion = 20;
 
 		int minorVersion = NMSUtil.getMinorVersion();
```
Comment (saharNooby) [inline src/main/java/me/saharnooby/plugins/leadwires/tracker/ProtocolUtil.java:110]:
```suggestion
		int minorVersion = NMSUtil.getMinorVersion();

```

File: src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java [hunk 4]
```diff
@@ -85,7 +85,13 @@ public void onPacketSending(PacketEvent e) {
...
+                } else {
+                    coord = new ChunkCoord(packet.getIntegers().read(0), packet.getIntegers().read(1));
+                }
```
Comment (saharNooby) [inline src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java]: Please fix the indentation on this and the 5 lines above.

**Review by saharNooby — CHANGES_REQUESTED** [review]
Hi! Thanks for the contribution!

However, code style needs minor fixes. After it is fixed, I'll be ready to merge and update the plugin on SpigotMC.

I trust that you've tested the plugin locally (especially wire end positions), since I would not be able to.

**Comment by saharNooby** [comment]
Please also replace all occurrences of `1.1.9` with `1.1.10` -- at least, there are occurrences in [README.md](https://github.com/saharNooby/lead-wires/blob/master/README.md) and [plugin.yml](https://github.com/saharNooby/lead-wires/blob/master/src/main/resources/plugin.yml).

**Comment by LedariaTim1** [comment]
Hi, I have adopted your minor code changes, I actually overlooked the version in those places. Yes, I tested it with papermc in versions 1.20, 1.20.1 and 1.20.2.

File: src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java (hunk 4 above)
Comment (LedariaTim1) [reply src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java]: thats crazy, in my ide the indentation is correct :D 
image

**Review by saharNooby — APPROVED** [review]

**Comment by saharNooby** [comment]
Big thanks! I'll build and upload it to SpigotMC soon.

File: src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java (hunk 4 above)
Comment (saharNooby) [reply src/main/java/me/saharnooby/plugins/leadwires/chunk/LoadedChunkTracker.java]: It's tabs vs spaces :)

**Comment by Ceddix** [comment]
Thank you!! ❤️

**Pull request merged**